core/src/main/java/org/neo4j/ogm/typeconversion/AttributeConverters.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This product is licensed to you under the Apache License, Version 2.0 (the "License").
+ * You may not use this product except in compliance with the License.
+ *
+ * This product may include a number of subcomponents with
+ * separate copyright notices and license terms. Your use of the source
+ * code for these subcomponents is subject to the terms and
+ *  conditions of the subcomponent's license, as noted in the LICENSE file.
+ */
+package org.neo4j.ogm.typeconversion;
+
+import java.util.function.Function;
+
+/**
+ * This class serves as a holder for converters from a given type to another. The types of the converters must match.
+ * They only differ in the way they treat scalars, arrays and iterables.
+ *
+ * @author Michael J. Simons
+ */
+public class AttributeConverters {
+    public final AttributeConverter<?, ?> forScalar;
+    public final AttributeConverter<?, ?> forArray;
+    public final Function<String, AttributeConverter<?, ?>> forIterable;
+
+    static class Builder {
+        AttributeConverters hlp;
+
+        private Builder() {
+        }
+
+        static Builder forScalar(AttributeConverter<?, ?> c) {
+            Builder rv = new Builder();
+            rv.hlp = new AttributeConverters(c, null, null);
+            return rv;
+        }
+
+        static AttributeConverters onlyScalar(AttributeConverter<?, ?> c) {
+            return forScalar(c).hlp;
+        }
+
+        static AttributeConverters onlyArray(AttributeConverter<?, ?> c) {
+            return forScalar(null).array(c).hlp;
+        }
+
+        Builder array(final AttributeConverter<?, ?> c) {
+            this.hlp = new AttributeConverters(this.hlp.forScalar, c, null);
+            return this;
+        }
+
+        AttributeConverters andIterable(Function<String, AttributeConverter<?, ?>> c) {
+            return new AttributeConverters(this.hlp.forScalar, this.hlp.forArray, c);
+        }
+
+    }
+
+    private AttributeConverters(AttributeConverter<?, ?> forScalar,
+        AttributeConverter<?, ?> forArray, Function<String, AttributeConverter<?, ?>> forIterable) {
+        this.forScalar = forScalar;
```

Comment:
Please a blank line between more than one line method signature and the inner content.